src/java/com/puppetlabs/puppetserver/ShellUtils.java

```diff
@@ -14,23 +14,56 @@
 
 public class ShellUtils {
 
+    public static class ExecutionOptions {
```

Comment:
Big 👍 on this refactor.  I agree that the method signatures seemed like they might have continued to get out of hand if we had continued down the same road of adding more parameters / method overloads to the base `ShellUtils` class.